core/wiki/macros/CSS.tid

```diff
@@ -74,3 +74,47 @@ column-count: $columns$;
 \define if-background-attachment(text)
 <$reveal state="$:/themes/tiddlywiki/vanilla/settings/backgroundimage" type="nomatch" text="">$text$</$reveal>
 \end
+
+\define if-sticky-titles-calc-zindex()
+``
+[data-tiddler-title="``$(cssEscapedTitle)$``"] .tc-tiddler-title {
+	z-index: calc(500 + ``$(storyCount)$`` - ``$(tiddlerIndex)$``);
+}
+``
+\end
+
+\define if-sticky-titles()
+<$reveal state="$:/themes/tiddlywiki/vanilla/options/stickytitles" type="match" text="yes">
+
+``
+.tc-tiddler-title {
+	position: -webkit-sticky;
+	position: -moz-sticky;
+	position: -o-sticky;
+	position: -ms-sticky;
+	position: sticky;
+	top: 0px;
+	background: ``<<colour tiddler-background>>``;
+}
+``
+
+<$vars storyCount={{{ [list[$:/StoryList]count[]] }}}>
```

Comment:
This sequence is very cunning, and now I see why you were interested in the CSS escaping! Great stuff. I've tried and failed to come up with a pure CSS solution, this looks like the next best thing.

Reply:
Thanks @Jermolene ... I also couldn't figure out how to do it in css. I had already figured out this logic for applying z-indices so it was just a matter of doing this when I saw the issue...